src/MSBuildTaskHost/FileSystem/MSBuildTaskHostFileSystem.cs

```diff
@@ -21,6 +21,26 @@ public bool DirectoryEntryExists(string path)
             return NativeMethodsShared.FileOrDirectoryExists(path);
         }
 
+        public void ClearCaches() { }
+        public void WriteStatistics(TextWriter writer) { }
+
+        public FileAttributes GetAttributesNoMissingException(string path)
```

Comment:
This seems like an awkward API to expose

Reply:
Yea, it's pretty funky. I'll remove the part with the exception.